velox/exec/AggregateCompanionAdapter.cpp

```diff
@@ -542,6 +452,52 @@ bool CompanionFunctionsRegistrar::registerExtractFunction(
     VELOX_FAIL(
         "Original aggregation function {} not found: {}", originalName, name);
   };
+}
+
+bool CompanionFunctionsRegistrar::registerExtractFunctionWithSuffix(
+    const std::string& originalName,
+    const std::vector<AggregateFunctionSignaturePtr>& signatures,
+    bool overwrite) {
+  auto groupedSignatures =
+      CompanionSignatures::groupSignaturesByReturnType(signatures);
+  bool registered = false;
+  for (const auto& [type, signatureGroup] : groupedSignatures) {
+    auto extractSignatures =
+        CompanionSignatures::extractFunctionSignatures(signatureGroup);
+    if (extractSignatures.empty()) {
+      continue;
+    }
+
+    auto factory = getVectorFunctionFactory(originalName);
+    registered |= exec::registerStatefulVectorFunction(
+        CompanionSignatures::extractFunctionNameWithSuffix(originalName, type),
+        extractSignatures,
+        factory,
```

Comment:
nit: std::move(extractSignatures) and std::move(factory). Ditto for lines 503--504.

Reply:
Hi @kagamiori Thank you for your reply.  I've reconstructed it.